### PR TITLE
client: warn when using a deprecated client binary name

### PR DESCRIPTION
**Pull request by lzaoral**

### CODE REVIEW

File: osh/client/osh-cli [hunk 1]
```diff
@@ -21,9 +23,15 @@ class OshCommandContainer(kobo.client.ClientCommandContainer):
...
+    # warn when using deprecated client binary name
+    argv0_base = os.path.basename(sys.argv[0])
+    if argv0_base == 'covscan':
+        warnings.warn(f'Using {argv0_base} binary instead of osh-cli is deprecated!',
+                      DeprecationWarning)
+
     def print_warning(opt):
-        print(f"warning: forcing {conf[opt]} as {opt}, please update client configuration!",
-              file=sys.stderr)
+        warnings.warn(f"Forcing {conf[opt]} as {opt}, please update client configuration!",
```
Comment (kdudka) [inline osh/client/osh-cli:33]: I am not sure whether this change is really an improvement from end user's point of view.  Previously it printed:
```
warning: forcing https://cov01.lab.eng.brq2.redhat.com/osh/xmlrpc as HUB_URL, please update client configuration!
```
Now it prints:
```
/usr/bin/covscan:33: DeprecationWarning: Forcing https://cov01.lab.eng.brq2.redhat.com/osh/xmlrpc as HUB_URL, please update client configuration!
  warnings.warn(f"Forcing {conf[opt]} as {opt}, please update client configuration!",
```
The new variant looks to me as if Python was reporting me some problem with the **code** of `osh-cli` (such as using a deprecated API in its implementation).  The information actually important for the end user is hidden between unrelated location info and a code snippet.

**Review by kdudka — CHANGES_REQUESTED** [review]

File: osh/client/osh-cli (hunk 1 above)
Comment (lzaoral) [reply osh/client/osh-cli]: Addressed in  487da91b119e56c3b6b59ecbb152be78dd6673d2.

Comment (lzaoral) [reply osh/client/osh-cli]: The only change in the latest rebase was for `s|binary|executable|`.

**Review by kdudka — APPROVED** [review]
Looks good.  Thanks for the update!

**Pull request merged**